### PR TITLE
Improve Repo.insert_all/3 typespec

### PR DESCRIPTION
**Pull request by v0idpwn**
Makes it a bit more specific, and also easier to read

### CODE REVIEW

**Comment by v0idpwn** [comment]
Makes it a bit more specific, and also easier to look at. Had someone confused because they were trying to insert_all with %{String.t() => term}

File: lib/ecto/repo.ex [hunk 1]
```diff
@@ -1331,9 +1331,9 @@ defmodule Ecto.Repo do
   @doc group: "Schema API"
   @callback insert_all(
               schema_or_source :: binary | {binary, module} | module,
-              entries_or_query :: [map | [{atom, term | Ecto.Query.t}]] | Ecto.Query.t,
+              entries_or_query :: [%{atom => value} | Keyword.t(value)] | Ecto.Query.t,
               opts :: Keyword.t()
-            ) :: {integer, nil | [term]}
+            ) :: {integer, nil | [term]} when value: term | Ecto.Query.t()
```
Comment (v0idpwn) [inline lib/ecto/repo.ex:1336]: If you have a better name than `value`, tell me :P

**Review by josevalim — APPROVED** [review]

File: lib/ecto/repo.ex (hunk 1 above)
Comment (fuelen) [reply lib/ecto/repo.ex:1336]: what about `integer` -> `non_neg_integer` for all *_all functions?

Comment (v0idpwn) [reply lib/ecto/repo.ex:1336]: I think that's fair, @fuelen. Pr welcome :)

**Pull request merged**